In [42]:
import pandas as pd

In [43]:
artists_df_orig = pd.read_json("./data/artists.jsonl", lines=True)
sessions_df_orig = pd.read_json("./data/sessions.jsonl", lines=True)
tracks_df_orig = pd.read_json("./data/tracks.jsonl", lines=True)
users_df_orig = pd.read_json("./data/users.jsonl", lines=True)

In [44]:
artists_df = artists_df_orig.copy(deep=True)
for c in artists_df.columns:
    incomplete_rows = artists_df[artists_df[[c]].isnull().any(axis=1)]
    print(c, len(incomplete_rows))

id 0
name 0
genres 81


artists.jsonl:
- id i name nie mają brakujących wartości
- genres można uzupełnić na podstawie piosenek, braki w 81 rekordach 

In [45]:
sessions_df = sessions_df_orig.copy(deep=True)
# incomplete_rows = sessions_df[sessions_df[['id', 'name']].isnull().any(axis=1)]
# print("Incomplete Rows:")
# print(incomplete_rows)

print("Liczba rekordów", len(sessions_df))
for c in sessions_df.columns:
    incomplete_rows = sessions_df[sessions_df[[c]].isnull().any(axis=1)]
    print(c, len(incomplete_rows))

Liczba rekordów 3173490
session_id 0
timestamp 0
user_id 158691
track_id 565752
event_type 158853


In [46]:
sessions_df = sessions_df_orig.copy(deep=True)

print("Unikalne wartości event_type", sessions_df['event_type'].unique())
sessions_df = sessions_df[sessions_df['event_type'].isin(['skip', 'like', None, 'play'])]

print("Liczba rekordów", len(sessions_df))
for c in sessions_df.columns:
    incomplete_rows = sessions_df[sessions_df[[c]].isnull().any(axis=1)]
    print(c, len(incomplete_rows))

Unikalne wartości event_type ['play' 'advertisment' 'like' 'skip' None]
Liczba rekordów 2766358
session_id 0
timestamp 0
user_id 138255
track_id 158620
event_type 158853


Na 3173490 rekordów:

dużo rekordów (>20k) mają event_type = "advertisment" -> nie są potrzebne 


In [47]:
tracks_df = tracks_df_orig.copy(deep=True)
incomplete_rows = tracks_df[tracks_df[tracks_df.columns].isnull().any(axis=1)]
print("Incomplete Rows:")
print(incomplete_rows)

Incomplete Rows:
                           id  \
8                        None   
10     2WaYW84yWij5NSCpgSeU2R   
13     0gE63HwMeWgK30ONO4witM   
14                       None   
31     5ueyLj6e6oVaTY0KQ6yLaA   
...                       ...   
22377  3zYkiaVV22ZjeZPS1YrFRa   
22382                    None   
22403  3JJYIoJ5FyY9E0DGjr7SXF   
22404  54j7EaJPDmSZYcNYvLSJ78   
22407  0LcNMuOiULmxJK3bdHTfDF   

                                                    name  popularity  \
8      In The Wee Small Hours Of The Morning - Remast...        60.0   
10           You Make Me Feel So Young - Remastered 1998         NaN   
13                                                  None        57.0   
14                                            My Romance        55.0   
31                                                  None        60.0   
...                                                  ...         ...   
22377                                          Svartsjuk        52.0   
22382     

tracks.jsonl:
- id: 1161 linii z brakiem
- name: 1146 linii z brakiem
- id i name: w 64 liniach brakuje obu wartości
w sumie 2243 brakuje albo id albo name

bez id nie jesteśmy w stanie użyć danego utworu
bez nazwy nie możemy przedstawić utworu użytkownikowi

In [48]:
users_df = users_df_orig.copy(deep=True)
incomplete_rows = users_df[users_df[users_df.columns[:-1]].isnull().any(axis=1)]
print("Incomplete Rows:")
print(incomplete_rows)


Incomplete Rows:
     user_id                  name      city                 street  \
2        103        Kornelia Witka    Gdynia     al. Kościuszki 587   
30       131          Karol Tarach    Kraków        aleja Kreta 272   
49       150       Roksana Ścibior    Kraków    pl. Wojciecha 95/32   
69       170  Marcelina Nazarewicz  Szczecin    plac Popiełuszki 93   
87       188           Natan Gołas    Poznań     plac Działkowa 637   
..       ...                   ...       ...                    ...   
888      989          Bartek Damek  Szczecin          pl. Orkana 51   
916     1017          Kamila Zadka    Poznań          al. Fredry 97   
939     1040        Tadeusz Kołpak  Warszawa        al. Wiosenna 04   
957     1058      Kajetan Gmurczyk  Szczecin  pl. Głowackiego 55/52   
981     1082           Inga Wojtak   Wrocław  plac Warszawska 04/66   

                                     favourite_genres  premium_user   id  
2                                                None  

w 90 liniach brakuje albo favourite_genres albo premium_user
ale to nie powinno przeszkadzać w realizacji zadania rekomendacji

nie wiadomo co znaczy atrybut id, który pojawia się 55 razy i zawsze ma wartość -1